### ISSUE TITLE
Where can I find a discussion of the development plan?

### ISSUE THREAD

**Issue by JimBiard** [neo4j-contrib/spatial#144]
Hi.

I'd like to contribute towards developing Neo4j Spatial. Is there a place where I can see the development plan and get involved in the discussion? I see a few areas where improvements appear to be needed, but without a sense of the direction it is hard to know how to proceed.

Grace and peace,

Jim Biard

**Comment by Infiniverse** [comment]
Hi Jim, did you ever get an answer to this question?

**Comment by craigtaverner** [comment]
Jim had a direct discussion with me. We have plans to add native spatial support directly to Neo4j during 2015, but the exact schedule is not yet finalised. As things become clearer, we'll make announcements. In the meantime, if you have feedback on what you like/dislike about Neo4j Spatial, feel free to let us know, and we'll see what we can do.

**Comment by Infiniverse** [comment]
I think that I’m over most of the learning curve now, so that’s good. I’ve found it hard to get up to speed (I’ve probably read most blog entries on spatial at least several times each!).

I’ll definitely let you know of surprises as I come across them, and look forward to testing the more integrated approach as it happens.

Thanks,
Joe

**Comment by devourment77** [comment]
Craig,

This part really interests me: "We have plans to add native spatial support directly to Neo4j during 2015, but the exact schedule is not yet finalised. "

Do you mean "native" like schema indexes are baked in?  Maybe the possibility of adding a spatial index w/o the need for a plugin?  Without the need for "START" in cypher (and just "MATCH").

Thanks in advance,
Brandon

**Comment by craigtaverner** [comment]
Yes Brandon, that is exactly what we mean. Similar to schema indexes, using a similar 'CREATE' statement, and no need for plugins or legacy index (START), only MATCH. At this point it appears that early releases will only support point data and focus on proximity search use cases, as this is seen as the most common case. But the spatial features will expand over time to cover further use cases. There are, however, some things that remain most suitable to an external library, and so the existing library will probably remain for those cases.

**Comment by rickystillwell** [comment]
Hi Craig,
I'm also very interested in this feature, proximity based search on a single point is fine for my use case right now... Do you have a target release for this new feature (even in beta)?

**Comment by cleishm** [comment]
Hey @rickystillwell,

The release plan for Neo4j itself is not something we typically discuss too far in advance, as it has a propensity to change and we like to avoid over-promising! But certainly the team has done quite a bit of thinking and planning in this area, so I'm hopeful we'll see some parts of it start to appear in the Neo4j 2.3 and later releases.

Cheers,
Chris
